src/ImageSharp/ColorSpaces/Conversion/Implementation/Icc/Calculators/ClutCalculator.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Numerics;
+using SixLabors.ImageSharp.MetaData.Profiles.Icc;
+
+namespace SixLabors.ImageSharp.ColorSpaces.Conversion.Implementation.Icc.Calculators
+{
+    internal class ClutCalculator : IVector4Calculator
+    {
+        private int inputCount;
+        private int outputCount;
+        private float[][] lut;
+        private byte[] gridPointCount;
+        private int[] indexFactor;
+        private int nodeCount;
+
+        public ClutCalculator(IccClut clut)
+        {
+            Guard.NotNull(clut, nameof(clut));
+
+            this.inputCount = clut.InputChannelCount;
+            this.outputCount = clut.OutputChannelCount;
+            this.lut = clut.Values;
+            this.gridPointCount = clut.GridPointCount;
+            this.indexFactor = this.CalculateIndexFactor(clut.InputChannelCount, clut.GridPointCount);
+            this.nodeCount = (int)Math.Pow(2, clut.InputChannelCount);
+        }
+
+        public Vector4 Calculate(Vector4 value)
+        {
+            Vector4.Clamp(value, Vector4.Zero, Vector4.One);
+
+            float[] result;
```

Comment:
Since we know that the maximum length is 4 could we `stackalloc` the array and pass it as a sliced `Span<byte>` to the `Interpolate` method?

Reply:
I'm not sure what you mean with the `Span<byte>` part but yes, I could use stackalloc also at other places and in general decrease memory allocations. I'll keep this in mind when I continue to work on it.

Reply:
Ah yes that makes more sense. Thank you for the example, doing it like that is a lot better.